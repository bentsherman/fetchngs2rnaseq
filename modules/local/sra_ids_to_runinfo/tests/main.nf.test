nextflow_process {

    name "Test process: SRA_IDS_TO_RUNINFO"
    script "../main.nf"
    process "SRA_IDS_TO_RUNINFO"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = 'SRR13191702'
                input[1] = ''
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
