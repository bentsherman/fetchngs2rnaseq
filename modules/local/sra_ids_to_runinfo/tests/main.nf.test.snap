{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "SRR13191702.runinfo.tsv:md5,3a1be35781ca6e8a28d8fd4d2f3bbe85"
                ],
                "1": [
                    "versions.yml:md5,1c14442e9b494b586eafe41e77300fae"
                ],
                "tsv": [
                    "SRR13191702.runinfo.tsv:md5,3a1be35781ca6e8a28d8fd4d2f3bbe85"
                ],
                "versions": [
                    "versions.yml:md5,1c14442e9b494b586eafe41e77300fae"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-28T11:52:05.345153"
    }
}